tests/models/prism/test_modeling_prism.py

```diff
@@ -0,0 +1,463 @@
+        )
+        inputs_dict = prepare_prism_inputs_dict(model.config, input_ids, decoder_input_ids)
+        with torch.no_grad():
+            output = model(**inputs_dict)[0]
+        expected_shape = torch.Size((1, 19, 1280))
+        self.assertEqual(output.shape, expected_shape)
+        # change to expected output here
+        expected_slice = torch.tensor(
+            [[-0.1113, -0.0307, -0.5641], [-0.4884, -0.2435, -0.3268], [-0.3196, -0.2342, -1.3295]],
+            device=torch_device,
+        )
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=TOLERANCE))
+
+    def test_inference_head(self):
+        model = PrismForConditionalGeneration.from_pretrained("dariast/prism").to(torch_device)
+        # change to intended input
+        input_ids = _long_tensor(
+            [[85, 16370, 15443, 5, 160, 50, 11, 1814, 21392, 8, 14787, 8, 14, 747, 125, 3522, 15352, 6, 2]]
+        )
+        decoder_input_ids = _long_tensor(
+            [[2, 85, 43185, 25, 38476, 165, 5330, 1185, 269, 784, 9474, 21, 7142, 220, 13, 3051, 3079, 6, 2]]
+        )
+        inputs_dict = prepare_prism_inputs_dict(model.config, input_ids, decoder_input_ids)
+        with torch.no_grad():
+            output = model(**inputs_dict)[0]
+        expected_shape = torch.Size((1, 19, model.config.vocab_size))
+        self.assertEqual(output.shape, expected_shape)
+        # change to expected output here
+        expected_slice = torch.tensor(
+            [[-0.2506, -0.2493, 3.5705], [-0.1940, -0.2163, 3.7116], [0.2059, 0.1973, 5.9530]], device=torch_device
+        )
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=TOLERANCE))
+
+    def test_seq_to_seq_generation(self):
+        model = PrismForConditionalGeneration.from_pretrained("dariast/prism").to(torch_device)
+        tokenizer = PrismTokenizer.from_pretrained("dariast/prism", src_lang="fr", tgt_lang="en")
+
+        src_fr = [
+            "L'affaire NSA souligne l'absence totale de débat sur le renseignement",
+            "Selon moi, il y a deux niveaux de réponse de la part du gouvernement français.",
+            "Lorsque François Hollande téléphone à Barack Obama ou quand le ministre des affaires étrangères Laurent"
+            " Fabius convoque l'ambassadeur des Etats-Unis, ils réagissent à une vraie découverte, qui est celle de"
+            " l'ampleur de la surveillance américaine sur l'ensemble des communications en France.",
+        ]
+
+        # The below article tests that we don't add any hypotheses outside of the top n_beams
+        dct = tokenizer(src_fr, padding=True, return_tensors="pt")
+
+        hypotheses_batch = model.generate(
+            input_ids=dct["input_ids"].to(torch_device),
+            attention_mask=dct["attention_mask"].to(torch_device),
+            num_beams=5,
+            max_new_tokens=100,
+            forced_bos_token_id=tokenizer.get_lang_id("en"),
+        )
+
+        expected_en = [
```

Comment:
The initial Prism model prepends the target language ID at the beginning of the encoded string for the decoder, and the output of the Fraiseq Prism model includes language ID during the conversion of generated tokens into strings. Moreover, M2M100 doesn't have language token IDs in the main vocabulary. Target language IDs are added to the ```special tokens``` list and processed differently to Prism tokenizer. In the case of the Prism model, target language IDs should be taken from the main vocabulary and should not be added to the ```special tokens```. Due to this difference, the ```convert_tokens_to_string``` that was identical to M2M100 before was changed so that no target language ID is visible in the final converted sequence. As of now, the Prism output doesn't have the target language ID, e.g., <en>, at the beginning of the converted sequence as it was before.

Reply:
Hi @DaryaTereshchenko, thanks for your response. I didn't look at your answer in detail yet, but what I can tell you is that the target language IDs should definitely be treated as special tokens. Maybe this will also resolve the remaining issues with the tokenizer, such as the `test_tokenizer_decode_ignores_language_codes` below.

Look at https://huggingface.co/facebook/m2m100_418M/blob/main/special_tokens_map.json to see how the language IDs can be configured as special tokens. (This should happen in the model repo, not in the code, because different model checkpoints might have different languages.)

Reply:
Closing this because `expected_en` now doesn't include the target language IDs, which was my original request in this conversation.